common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -534,23 +534,40 @@ public boolean contentEqualsIgnoreCase(CharSequence string) {
         }
 
         if (string instanceof AsciiString) {
-            AsciiString rhs = (AsciiString) string;
-            for (int i = arrayOffset(), j = rhs.arrayOffset(), end = i + length(); i < end; ++i, ++j) {
-                if (!equalsIgnoreCase(value[i], rhs.value[j])) {
-                    return false;
+            AsciiString other = (AsciiString) string;
+            byte[] value = this.value;
+            if (offset == 0 && other.offset == 0 && length == value.length) {
```

Comment:
Eheh I know, I have already spoken with the OpenJDK engs in Red hat, because I am surprised as well, but I have verified that this optimization work across all the versions I have tested (11, 17, 21, main); I will add a comment if it can help, because I understand that is help to correlate the length of the loop with the array boundaries, to fully remove the value bound checks (not the other one, sadly, which would requires unsafe access to do it, instead)

Reply:
I've tried your suggestion and getting the old numbers ie ~180 M ops/sec, vs current one, which are ~200/210 M req/sec. Given that most of the time `length == value.length` I don't feel good to loose that perf boost, given that will still cover most of the cases...